packages/components/segmented/src/segmented.vue

```diff
@@ -159,7 +159,7 @@ useResizeObserver(segmentedRef, updateSelect)
 watch(activeElement, updateSelect)
 
 watch(
-  () => props.modelValue,
+  [() => props.modelValue, () => props.disabled],
```

Comment:
Looks like this issues with the loss of reactive caused by [Line 128](https://github.com/element-plus/element-plus/pull/17716/files#diff-497b5081ac7ccf558d671f5f2aeb9819d52a2eada9c8a3d8892f32f62cd92f08L128).

Forced updates will cost more performance

Reply:
Maybe `state.disabled` -> `getDisabled(getOption(props.modelValue))`

I feel like this way we won't lose reactive.

Reply:
I think watch `props.disabled` is ok.